api/utils/requestProcessor.js

```diff
@@ -2148,6 +2202,132 @@ const validateAppForWriteAPI = (params, done, try_times) => {
     });
 };
 
+/**
+ * Validate app for fetch API from sdk
+ * @param  {object} params - params object
+ * @param  {function} done - callback when processing done
+ */
+const validateAppForFetchAPI = (params, done) => {
+    ignorePossibleDevices(params, done);
+
+    common.db.collection('apps').findOne({'key': params.qstring.app_key}, (err, app) => {
+        if (!app) {
+            common.returnMessage(params, 400, 'App does not exist');
+            return done ? done() : false;
+        }
+
+        params.app_id = app._id;
+        params.app_cc = app.country;
+        params.app_name = app.name;
+        params.appTimezone = app.timezone;
+        params.app = app;
+        params.time = common.initTimeObj(params.appTimezone, params.qstring.timestamp);
+
+        if (params.app.checksum_salt && params.app.checksum_salt.length) {
```

Comment:
Maybe we can have single method for checksum check for both write and read functions?
To avoid code duplication etc

Reply:
done.